Snowflake.Data.Tests/Mock/MockAzureClient.cs

```diff
@@ -0,0 +1,166 @@
+
+            return azureError;
+        }
+
+        internal Response<BlobProperties> createMockResponseForBlobProperties()
+        {
+            if (_key == HttpStatusCode.OK.ToString())
+            {
+                Dictionary<string, string> metadata = new Dictionary<string, string>
+                {
+                    { "encryptiondata",
+                        "{" +
+                        $"\"ContentEncryptionIV\": \"{AzureIV}\", " +
+                        $"\"WrappedContentKey\": {{\"EncryptedKey\":\"{AzureKey}\"}}" +
+                        "}"
+                    },
+                    {
+                        "matdesc", AzureMatdesc
+                    },
+                    {
+                        "sfcdigest", SFCDigest
+                    }
+                };
+
+                BlobProperties blobProperties = BlobsModelFactory.BlobProperties(metadata: metadata, contentLength: ContentLength);
+
+                Response<BlobProperties> mockResponse = Response.FromValue(
+                            blobProperties,
+                            null);
+
+                return mockResponse;
+            }
+            else
+            {
+                throw CreateMockAzureError(_key);
+            }
+        }
+
+        public override Response<BlobProperties> GetProperties(
+            BlobRequestConditions conditions = default,
+            CancellationToken cancellationToken = default)
+        {
+            return createMockResponseForBlobProperties();
+        }
+
+        public override async Task<Response<BlobProperties>> GetPropertiesAsync(
+            BlobRequestConditions conditions = default,
+            CancellationToken cancellationToken = default)
+        {
+            return await Task.Run(() => createMockResponseForBlobProperties()).ConfigureAwait(false);
+        }
+
+        public Response<BlobContentInfo> createMockResponseForBlobContentInfo()
+        {
+            if (_key == HttpStatusCode.OK.ToString())
+            {
+                return null;
+            }
+            else
+            {
+                throw CreateMockAzureError(_key);
+            }
+        }
+
+        public override Response<BlobContentInfo> Upload(Stream content)
+        {
+            return createMockResponseForBlobContentInfo();
+        }
+        
+        public override async Task<Response<BlobContentInfo>> UploadAsync(Stream content,
+            CancellationToken cancellationToken = default)
+        {
+            return await Task.Run(() => createMockResponseForBlobContentInfo()).ConfigureAwait(false);
+        }
+
+        public override Response<BlobInfo> SetMetadata(
+            Metadata metadata,
+            BlobRequestConditions conditions = default,
+            CancellationToken cancellationToken = default)
+        {
+            return null;
+        }
+
+        public override Response DownloadTo(string path)
```

Comment:
instead of creating a mock class and putting logic into it you could try to use Moq librabary to mock the BlobClient. I mean in the test it would look like something like that:

var mock = new Mock<BlobClient>();
mock.CallBase = true;
mock.Setup(client => client.DownloadTo(...))
         .Returns(something-to-return)
so in the test you could write what a given method should return.

Reply:
I refactored the mock class using Moq